### PR TITLE
chore: Fix `inx` and `api` features

### PR DESCRIPTION
**Pull request by Alexandcoats**
Fixed the CI defs so that the "common" feature groups are more accurate. Did some blanket testing of feature combos as well.

### CODE REVIEW

File: bin/inx-chronicle/src/stardust_inx/error.rs [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 use thiserror::Error;
 
-use super::listener::InxListenerError;
+use super::listener::StardustInxListenerError;
```
Comment (grtlr) [inline bin/inx-chronicle/src/stardust_inx/error.rs:6]: I wonder if that should just be `stardust::InxListenerError`?

File: bin/inx-chronicle/src/config.rs [hunk 2]
```diff
@@ -9,8 +9,8 @@ use thiserror::Error;
 
 #[cfg(feature = "api")]
 use crate::api::ApiConfig;
-#[cfg(feature = "inx")]
-use crate::inx::InxConfig;
+#[cfg(all(feature = "stardust", feature = "inx"))]
+use crate::stardust_inx::StardustInxConfig;
```
Comment (grtlr) [inline bin/inx-chronicle/src/config.rs:13]: Same here (maybe `stardust::InxConfig`)?

File: bin/inx-chronicle/src/stardust_inx/error.rs (hunk 1 above)
Comment (Alexandcoats) [reply bin/inx-chronicle/src/stardust_inx/error.rs:6]: I renamed them so we wouldn't have potential naming conflicts later in the event that we do add more protocols that use INX.

**Comment by grtlr** [comment]
> I renamed them so we wouldn't have potential naming conflicts later in the event that we do add more protocols that use INX. 

Yeah, that's also fine. I guess if we ever want to support other protocols, we could think about defining a trait that abstracts over that anyways.

**Review by grtlr — DISMISSED** [review]

**Comment by grtlr** [comment]
Superseded by #128.

**Review by grtlr — APPROVED** [review]

**Pull request merged**